plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/procedure/MigrateProcedure.java

```diff
@@ -297,24 +293,17 @@ private Schema toIcebergSchema(List<Column> columns)
         List<Types.NestedField> icebergColumns = new ArrayList<>();
         for (Column column : columns) {
             int index = icebergColumns.size();
-            org.apache.iceberg.types.Type type = toIcebergType(typeManager.getType(column.getType().getTypeSignature()), nextFieldId);
+            org.apache.iceberg.types.Type type = toIcebergTypeForNewColumn(typeManager.getType(column.getType().getTypeSignature()), nextFieldId);
             Types.NestedField field = Types.NestedField.of(index, false, column.getName(), type, column.getComment().orElse(null));
             icebergColumns.add(field);
         }
+
+        nextFieldId.set(1);
```

Comment:
This "reset" operation is rather hard to follow.

Ideally we shouldn't have the following call of `TypeUtil.assignFreshIds(...)` because we've already done the assignment previously.
At the moment, the IDs which are assigned in the loop are fully discarded. At the very least, I'd recommend using for a better readability (even a comment would be welcomed):

```
icebergSchema = TypeUtil.assignFreshIds(icebergSchema, new AtomicInteger(1)::getAndIncrement);
```


I've did 

```
    private Schema toIcebergSchema(List<Column> columns)
    {
        AtomicInteger nextFieldId = new AtomicInteger(1);
        List<Types.NestedField> icebergColumns = new ArrayList<>();
        for (Column column : columns) {
            Types.NestedField field = Types.NestedField.of(nextFieldId.getAndIncrement(), false, column.getName(), toIcebergTypeForNewColumn(typeManager.getType(column.getType().getTypeSignature()), nextFieldId), column.getComment().orElse(null));
            icebergColumns.add(field);
        }

        org.apache.iceberg.types.Type icebergSchema = Types.StructType.of(icebergColumns);
        return new Schema(icebergSchema.asStructType().fields());
    }
```

However the tests fail, because, even though the fields get unique ids, `TypeUtil.assignFreshIds` works in a more breadth first fashion while the current logic of assigning IDs (via `toIcebergTypeForNewColumn` is depth-first.

Reply:
Thanks for the point, I have thought about using breadth first way to build the columns, but feels a little bit overkill for this method and the readability may not even better, WDYT?